sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -130,6 +130,10 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
     //   since the other rules might make two separate Unions operators adjacent.
     Batch("Union", Once,
       CombineUnions) ::
+    // run this once earlier. this might simplify the plan and reduce cost of optimizer
```

Comment:
Hi @bogdanrdc , could explain more about this?  Why is it necessary?

Reply:
it makes the optimizer faster for short queries. see code above. a query such as `Filter(LocalRelation)`, without this change, would go through all the heavy optimizer rules. with this change, the query becomes just `LocalRelation` which doesn't trigger many rules.

Reply:
can you put your comment above into the comment in code